munit/shared/src/main/scala/munit/Assertions.scala

```diff
@@ -189,14 +189,15 @@ trait Assertions extends MacroCompat.CompileErrorMacro {
       expectedExceptionMessage: Option[String],
       body: => Any
   )(implicit T: ClassTag[T], loc: Location): T = {
+    val expectedExceptionMsg =
+      s"expected exception of type '${T.runtimeClass.getName()}' but body evaluated successfully"
     try {
       body
-      fail(
-        s"expected exception of type '${T.runtimeClass.getName()}' but body evaluated successfully"
-      )
+      fail(expectedExceptionMsg)
     } catch {
       case e: FailExceptionLike[_]
-          if !T.runtimeClass.isAssignableFrom(e.getClass()) =>
+          if !T.runtimeClass.isAssignableFrom(e.getClass()) ||
+            e.getMessage.contains(expectedExceptionMsg) =>
```

Comment:
maybe a separate case:
```
```suggestion
           case failed: FailException if  failed.getMessage() == expectedExceptionMsg  =>
             throw e
```
to be super sure? So that we only employ this logic if someone throws exactly the same error as we would. Then they're on their own

Reply:
I was caught by this AssertionError bug today, and found this fix. @tgodzik 's suggestion is good, but 
```
failed.getMessage() == expectedExceptionMsg
```
need to be changed to
```
failed.getMessage.contains(expectedExceptionMsg)
```
Tested this and @mzuehlke 's original change, both worked.

Can this be merged?

Reply:
I totally forgot about this, @mzuehlke what do you think about suggested changes? I am ok also merging as is, we could update the test case though.

Reply:
I had forgotten about it, too. Sorry for that.
I adopted your suggestion.